### PR TITLE
refactor: enable -Wconversion warning for lua/treesitter.c

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/567

### CODE REVIEW

**Comment by justinmk** [comment]
Just curious, how did this get ignored? New files should be subject to -Wconversion from the start.

**Comment by dundargoc** [comment]
> Just curious, how did this get ignored? New files should be subject to -Wconversion from the start.

It was obviously manually added to the conversion ignore list. Or do you mean how it slipped past code review?

File: src/nvim/lua/treesitter.c [hunk 1]
```diff
@@ -208,17 +208,17 @@ int tslua_inspect_lang(lua_State *L)
 
   lua_createtable(L, 0, 2);  // [retval]
 
-  size_t nsymbols = (size_t)ts_language_symbol_count(lang);
+  int nsymbols = (int)ts_language_symbol_count(lang);
 
   lua_createtable(L, nsymbols - 1, 1);  // [retval, symbols]
-  for (size_t i = 0; i < nsymbols; i++) {
```
Comment (vigoux) [inline src/nvim/lua/treesitter.c:214]: I would rather keep the `size_t` here if possible.

File: src/nvim/lua/treesitter.c [hunk 2]
```diff
@@ -227,11 +227,11 @@ int tslua_inspect_lang(lua_State *L)
 
   lua_setfield(L, -2, "symbols");  // [retval]
 
-  size_t nfields = (size_t)ts_language_field_count(lang);
+  int nfields = (int)ts_language_field_count(lang);
   lua_createtable(L, nfields, 1);  // [retval, fields]
   // Field IDs go from 1 to nfields inclusive (extra index 0 maps to NULL)
-  for (size_t i = 1; i <= nfields; i++) {
-    lua_pushstring(L, ts_language_field_name_for_id(lang, i));
+  for (int i = 1; i <= nfields; i++) {
```
Comment (vigoux) [inline src/nvim/lua/treesitter.c:233]: Same here keeping `size_t`

File: src/nvim/lua/treesitter.c [hunk 3]
```diff
@@ -322,10 +322,10 @@ static const char *input_cb(void *payload, uint32_t byte_index, TSPoint position
 #undef BUFSIZE
 }
 
-static void push_ranges(lua_State *L, const TSRange *ranges, const unsigned int length)
+static void push_ranges(lua_State *L, const TSRange *ranges, const int length)
 {
   lua_createtable(L, length, 0);
-  for (size_t i = 0; i < length; i++) {
+  for (int i = 0; i < length; i++) {
```
Comment (vigoux) [inline src/nvim/lua/treesitter.c:328]: Same here with `size_t`

File: src/nvim/lua/treesitter.c [hunk 4]
```diff
@@ -793,7 +792,7 @@ static int node_field(lua_State *L)
   TSTreeCursor cursor = ts_tree_cursor_new(node);
 
   lua_newtable(L);  // [table]
-  unsigned int curr_index = 0;
+  int curr_index = 0;
```
Comment (vigoux) [inline src/nvim/lua/treesitter.c:795]:
```suggestion
  size_t curr_index = 0;
```

**Review by vigoux — COMMENTED** [review]
As discussed on matrix, I am going to make the changes

**Review by vigoux — APPROVED** [review]

File: src/nvim/lua/treesitter.c [hunk 5]
```diff
@@ -1212,6 +1217,8 @@ static const char *query_err_string(TSQueryError err)
     return "invalid field";
   case TSQueryErrorCapture:
     return "invalid capture";
+  case TSQueryErrorStructure:
+    return "invalid structur";
```
Comment (dundargoc) [inline src/nvim/lua/treesitter.c:1221]: *structure

Comment (vigoux) [reply src/nvim/lua/treesitter.c:1221]: Oh, right

**Comment by dundargoc** [comment]
LGTM.

**Pull request merged**